app/src/main/java/org/mozilla/fenix/home/recenttabs/view/RecentTabViewDecorator.kt

```diff
@@ -0,0 +1,132 @@
+     */
+    SINGLE,
+
+    /**
+     * This item is to be shown at the top of the section with others below it.
+     */
+    TOP,
+
+    /**
+     * This item is to be shown between others in this section.
+     */
+    MIDDLE,
+
+    /**
+     * This item is to be shown at the bottom of the section with others above it.
+     */
+    BOTTOM
+}
+
+/**
+ * Helpers for setting various layout properties for the view from a [RecentTabViewHolder].
+ *
+ * Depending on the provided [RecentTabsItemPosition]:
+ * - set a different background so that the entire section possibly containing
+ * more such items would have rounded corners but sibling items not.
+ * - sets small margins for the items so that there's a clear separation between siblings
+ */
+sealed class RecentTabViewDecorator {
+    /**
+     * Apply the decoration to [itemView].
+     */
+    abstract operator fun invoke(itemView: View): View
+
+    companion object {
+        /**
+         * Get the appropriate decorator to set view background / margins depending on the position
+         * of that view in the recent tabs section.
+         */
+        fun forPosition(position: RecentTabsItemPosition) = when (position) {
+            RecentTabsItemPosition.SINGLE -> SingleTabDecoration
+            RecentTabsItemPosition.TOP -> TopTabDecoration
+            RecentTabsItemPosition.MIDDLE -> MiddleTabDecoration
+            RecentTabsItemPosition.BOTTOM -> BottomTabDecoration
+        }
+    }
+
+    /**
+     * Decorator for a view shown in the recent tabs section that will update it to express
+     * that that item is the single one shown in this section.
+     */
+    object SingleTabDecoration : RecentTabViewDecorator() {
+        override fun invoke(itemView: View): View {
+            val context = itemView.context
+
+            itemView.background =
+                AppCompatResources.getDrawable(context, R.drawable.home_list_row_background)
+
+            return itemView
+        }
+    }
+
+    /**
+     * Decorator for a view shown in the recent tabs section that will update it to express
+     * that this is an item shown at the top of the section and there are others below it.
+     */
+    object TopTabDecoration : RecentTabViewDecorator() {
+        override fun invoke(itemView: View): View {
+            val context = itemView.context
+
+            itemView.background =
+                    AppCompatResources.getDrawable(context, R.drawable.rounded_top_corners)
+
+            return itemView
+        }
+    }
+
+    /**
+     * Decorator for a view shown in the recent tabs section that will update it to express
+     * that this is an item shown has other recents tabs to be shown on top or below it.
+     */
+    object MiddleTabDecoration : RecentTabViewDecorator() {
+        override fun invoke(itemView: View): View {
+            val context = itemView.context
+
+            itemView.setBackgroundColor(context.getColorFromAttr(R.attr.above))
```

Comment:
There is a different background for each item depending on the position in the "Jump back in section".
In all other cases we use a drawable with rounded corners.
For a middle item we don't need rounded corners, just the right color.

[top, middle, middle, bottom] items would look like:

![image](https://user-images.githubusercontent.com/11428869/124632347-ebee9000-de8c-11eb-8be7-4faf5d6e2d7d.png)

Reply:
Doesn't the colour stay the same rounded or not? I would have expected the background drawable to have only changed here. If I'm misunderstanding, please ignore.

Reply:
I went with not setting any background for the items in xml and then setting a different one for each variation in code.
If not setting a background here this would be the results:
| dark theme | light theme |
| ----------- | ----------- |
| ![image](https://user-images.githubusercontent.com/11428869/125250666-35acff80-e2ff-11eb-8ec4-97f991c5fb0f.png) | ![image](https://user-images.githubusercontent.com/11428869/125250779-4f4e4700-e2ff-11eb-83d5-c53b1703db88.png) |

I find it cleaner and more helpful having all customization used in one file making it easy to spot the differences.
Can also set that color as `background` in xml and then do nothing here but overwrite it in all other cases.
Or are you proposing a yet different approach?

Reply:
Having `android:src="?attr/above"` set in xml and then use 
> AppCompatResources.getDrawable(context, R.drawable.rounded_bottom_corners)

in code would work but the current `R.drawable.rounded_bottom_corners` (& co) are shapes also declaring a background color so to keep the implementation clean we'd need to create new shapes just for the rounded corners while allowing the colors to be set separately.

Do you propose going on this route?

Reply:
> I find it cleaner and more helpful having all customization used in one file making it easy to spot the differences.

I agree! I hope we can get this right when we migrate to Compose.

> Do you propose going on this route?

No, I think this is fine. Thank you for your detailed explanation. 🙂